sandbox/executor.cc

```diff
@@ -0,0 +1,175 @@
+#include "sandboxed_api/sandbox2/policy.h"
+#include "sandboxed_api/sandbox2/policybuilder.h"
+#include "sandboxed_api/sandbox2/result.h"
+#include "sandboxed_api/sandbox2/sandbox2.h"
+#include "sandboxed_api/sandbox2/util.h"
+#include "sandboxed_api/sandbox2/util/bpf_helper.h"
+#include "sandboxed_api/util/flag.h"
+#include "tools/cpp/runfiles/runfiles.h"
+
+std::unique_ptr<sandbox2::Policy> GetPolicy(std::string nodePath) {
+  return sandbox2::PolicyBuilder()
+      // The most frequent syscall should go first in this sequence (to make it
+      // fast).
+      // Allow read() with all arguments.
+      .AllowRead()
+      .AllowWrite()
+      // Allow a preset of syscalls that are known to be used during startup
+      // of static binaries.
+      .AllowDynamicStartup()
+      .AddLibrariesForBinary(nodePath)
+      .EnableNamespaces()
+      // Allow the getpid() syscall.
+      .AllowSyscall(__NR_getpid)
+
+      .AddDirectory("/proc")
+      .AddDirectory("/dev")
+      // .AddFileAt("/dev/zero", "/dev/fd/1022", false)
+
+      // Examples for AddPolicyOnSyscall:
+      .AddPolicyOnSyscall(__NR_write,
+                          {
+                              // Load the first argument of write() (= fd)
+                              ARG_32(0),
+                              // Allow write(fd=STDOUT)
+                              JEQ32(1, ALLOW),
+                              // Allow write(fd=STDERR)
+                              JEQ32(2, ALLOW),
+                          })
+      // write() calls with fd not in (1, 2) will continue evaluating the
+      // policy. This means that other rules might still allow them.
+
+      // Allow exit() only with an exit_code of 0.
+      // Explicitly jumping to KILL, thus the following rules can not
+      // override this rule.
+      .AddPolicyOnSyscall(
+          __NR_exit_group,
+          {// Load first argument (exit_code).
+           ARG_32(0),
+           // Deny every argument except 0.
+           JNE32(0, KILL),
+           // Allow all exit() calls that were not previously forbidden
+           // = exit_code == 0.
+           ALLOW})
+
+      // = This won't have any effect as we handled every case of this syscall
+      // in the previous rule.
+      .AllowSyscall(__NR_exit_group)
+
+#ifdef __NR_open
+      .BlockSyscallWithErrno(__NR_open, ENOENT)
+#else
+      .BlockSyscallWithErrno(__NR_openat, ENOENT)
+#endif
+      .BuildOrDie();
+}
+
+void OutputFD(int stdoutFd, int errFd) {
+  for (;;) {
+    char stdoutBuf[4096];
+    char stderrBuf[4096];
+    ssize_t stdoutRLen = read(errFd, stdoutBuf, sizeof(stdoutBuf));
+    ssize_t stderrRLen = read(errFd, stderrBuf, sizeof(stderrBuf));
+    printf("stdout: '%s'\n", std::string(stdoutBuf, stdoutRLen).c_str());
+    printf("stderr: '%s'\n", std::string(stderrBuf, stderrRLen).c_str());
+    if (stdoutRLen < 1) {
+      break;
+    }
+  }
+}
+
+std::string GetDataDependencyFilePath(absl::string_view relative_path) {
```

Comment:
Sandboxed API 19fd11b91e6678db2fcfe69dd4037605730f5317 made `util/runfiles.h` public, so you may want to update the dependency.

Reply:
Awesome thanks